File: ts/src/platform/worker_app.dart
library angular2.src.platform.worker_app;

import 'package:angular2/src/core/zone/ng_zone.dart';
import 'package:angular2/src/platform/server/webworker_adapter.dart';
import 'package:angular2/src/platform/worker_app_common.dart';
import 'package:angular2/src/web_workers/shared/isolate_message_bus.dart';
import 'dart:isolate';

setupIsolate(SendPort replyTo) {
  return (NgZone zone) {
    WebWorkerDomAdapter.makeCurrent();

    ReceivePort rPort = new ReceivePort();
    var sink = new WebWorkerMessageBusSink(replyTo, rPort);
    var source = new IsolateMessageBusSource(rPort);
    IsolateMessageBus bus = new IsolateMessageBus(sink, source);
    return genericWorkerAppProviders(bus, zone);
  };
}

class WebWorkerMessageBusSink extends IsolateMessageBusSink {
  WebWorkerMessageBusSink(SendPort sPort, ReceivePort rPort) : super(sPort) {
    sPort.send(rPort.sendPort);
  }
}

File: ts/platform/worker_app.ts
export {
  WORKER_APP_PLATFORM,
  WORKER_APP_APPLICATION_COMMON
} from 'angular2/src/platform/worker_app_common';
export {WORKER_APP_APPLICATION} from 'angular2/src/platform/worker_app';
export {
  ClientMessageBroker,
  ClientMessageBrokerFactory,
  FnArg,
  UiArguments
} from '../src/web_workers/shared/client_message_broker';
export {
  ReceivedMessage,
  ServiceMessageBroker,
  ServiceMessageBrokerFactory
} from '../src/web_workers/shared/service_message_broker';
export {PRIMITIVE} from '../src/web_workers/shared/serializer';
export * from '../src/web_workers/shared/message_bus';
export {AngularEntrypoint} from 'angular2/src/core/angular_entrypoint';
